### ISSUE TITLE
Reduce failure due to large second dimension

### ISSUE THREAD

**Issue by mlloreda** [arrayfire/arrayfire#1702]
First reported [here](https://groups.google.com/forum/?utm_medium=email&utm_source=footer#!msg/arrayfire-users/I8g-Ec9oK8s/UIxaZAhoEgAJ).

Failure is due to exceeding launch configuration limits on CUDA. This appears to be related to issue #133.

Code to reproduce

```
TEST(Reduce, ISSUE_1702)
{
  af::array A = af::randu(1, 65535*8 + 1);
  af::array B = af::sum(A, 0);
  SUCCEED();
}
```

Error message:

```
Running main() from gtest_main.cc
Note: Google Test filter = *ISSUE_1702*
[==========] Running 1 test from 1 test case.
[----------] Global test environment set-up.
[----------] 1 test from Reduce
[ RUN      ] Reduce.ISSUE_1702
unknown file: Failure
C++ exception with description "ArrayFire Exception (Internal error:998):
In function void cuda::kernel::reduce_first_launcher(cuda::Param<To>, cuda::CParam<T>, cuda::uint, cuda::uint, cuda::uint, bool, double) [with Ti = float; To = float; af_op_t op = (af_op_t)0u; cuda::uint = uns\
igned int]
In file src/backend/cuda/kernel/reduce.hpp:328
CUDA Error (9): invalid configuration argument


In function af::array af::sum(const af::array&, int)
In file src/api/cpp/reduce.cpp:21" thrown in the test body.
[  FAILED  ] Reduce.ISSUE_1702 (694 ms)
[----------] 1 test from Reduce (694 ms total)

[----------] Global test environment tear-down
[==========] 1 test from 1 test case ran. (694 ms total)
[  PASSED  ] 0 tests.
[  FAILED  ] 1 test, listed below:
[  FAILED  ] Reduce.ISSUE_1702

 1 FAILED TEST

```

**Comment by pavanky** [comment]
Not a bug in reduce. Bug in JIT. Duplicate of #133

**Comment by pavanky** [comment]
@mlloreda reopen the issue if it fails with randu instead of constant.

**Comment by mlloreda** [comment]
@pavanky, failing with `randu` as well.

**Comment by mlloreda** [comment]
@pavanky done.

**Comment by pavanky** [comment]
Fixed in v3.5
